src/tuples-and-arrays/arrays.md

```diff
@@ -27,8 +27,8 @@ fn main() {
   later.
 
 - Try accessing an out-of-bounds array element. Array accesses are checked at
-  runtime. Rust can usually optimize these checks away, and they can be avoided
-  using unsafe Rust.
+  compile time. Rust can usually optimize these checks away, and they can be
```

Comment:
The original meaning was correct, and important.  In more sophisticated cases than this simple example, Rust may very well leave the runtime bounds-checks in place, and these can result in a substantial performance overhead above unchecked languages like C++.

It's difficult to demonstrate this in a short code sample because the checks are often optimized away at compile time -- in this case to an unconditional panic.  I suspect [`black_box`](https://doc.rust-lang.org/stable/std/hint/fn.black_box.html) could be useful to avoid this optimization. Do you want to try adjusting the example so that it shows the runtime bounds check?

Reply:
Ah I see what you mean. I tried using `black_box` but it seems to not prevent the compile-time safety checks. I changed the out-of-bound index to be returned from a function and add that example to below the note. PTAL. Let me know what you think.

Reply:
Oops. PTAL again.